.github/workflows/src/arm-auto-signoff/trivial-changes-check.js

```diff
@@ -0,0 +1,459 @@
+    );
+    return true;
+  }
+
+  // Deletions of docs/examples are OK, but already filtered above
+  return false;
+}
+
+/**
+ * Analyzes a PR to determine what types of changes it contains and updates the changes object
+ * @param {string[]} changedFiles - Array of changed file paths
+ * @param {import('simple-git').SimpleGit} git - Git instance
+ * @param {typeof import("@actions/core")} core - Core logger
+ * @param {import('./pr-changes.js').PullRequestChanges} changes - Changes object to update
+ * @returns {Promise<void>}
+ */
+async function analyzePullRequestChanges(changedFiles, git, core, changes) {
+  // Categorize files by type
+  const documentationFiles = changedFiles.filter(markdown);
+  const exampleFiles = changedFiles.filter(example);
+  const specFiles = changedFiles.filter(json).filter((file) => !example(file));
+  const otherFiles = changedFiles.filter((file) => !json(file) && !markdown(file));
+
+  core.info(
+    `File breakdown: ${documentationFiles.length} docs, ${exampleFiles.length} examples, ${specFiles.length} specs, ${otherFiles.length} other`,
+  );
+
+  // Set flags for file types present
+  if (documentationFiles.length > 0) {
+    changes.documentation = true;
+  }
+
+  if (exampleFiles.length > 0) {
+    changes.examples = true;
+  }
+
+  if (otherFiles.length > 0) {
+    changes.other = true;
+  }
+
+  // Analyze spec files to determine if functional or non-functional
+  if (specFiles.length > 0) {
+    core.info(`Analyzing ${specFiles.length} spec files...`);
+
+    let hasFunctionalChanges = false;
+
+    for (const file of specFiles) {
+      core.info(`Analyzing file: ${file}`);
+
+      try {
+        const isNonFunctional = await analyzeSpecFileForNonFunctionalChanges(file, git, core);
+        if (isNonFunctional) {
+          core.info(`File ${file} contains only non-functional changes`);
+        } else {
+          hasFunctionalChanges = true;
+          core.info(`File ${file} contains functional changes`);
+          // Once we find functional changes, we can stop
+          break;
+        }
+      } catch (error) {
+        core.warning(`Failed to analyze ${file}: ${getErrorMessage(error)}`);
+        // On error, treat as functional to be conservative
+        hasFunctionalChanges = true;
+        break;
+      }
+    }
+
+    changes.functional = hasFunctionalChanges;
+  }
+}
+
+/**
+ * Analyzes changes in a spec file to determine if they are only non-functional
+ * Note: New files and deletions should already be filtered by checkForNonTrivialFileOperations
+ * @param {string} file - The file path
+ * @param {import('simple-git').SimpleGit} git - Git instance
+ * @param {typeof import("@actions/core")} core - Core logger
+ * @returns {Promise<boolean>} - True if changes are non-functional only, false if any functional changes detected
+ */
+async function analyzeSpecFileForNonFunctionalChanges(file, git, core) {
```

Comment:
```suggestion
async function containsFunctionalChanges(file, git, core) {
```

This name is clearer and avoids double-negatives (but requires inverting the return values in code).